viewing_party/party.py

```diff
@@ -1,23 +1,173 @@
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+        movie_dict = {}
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+    else:
+        return None
+    
+    return movie_dict
+
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+
+def watch_movie(user_data, title):
+
+#loop through the dictionaries in the list user_data["watchlist"]
+#if the dictionary key == "title", check if the value == title (parameter)
+#if the value == title, save the dictionary to user_data["watched"]. Then delete the dictionary from user_data["watchlist"]
+
+
+    for dict in user_data["watchlist"]:
+        if dict["title"] == title:
+            user_data["watched"].append(dict)
+            user_data["watchlist"].remove(dict)
+
+    return user_data
+
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
+
+def get_watched_avg_rating(user_data):
+
+    ratings_list = []

```

Comment:
If `user_data["watched"]` doesn't hold anything, then we declare the variable `rating_list` but it never gets used. If we handle the invalid data at the very top of the function with it's own if-statement, we can both move the rest of the code out of if/else blocks and be sure we are allocating variables where we know they will be used.

Reply:
This makes sense!